android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt

```diff
@@ -183,38 +211,123 @@ class ProfileViewModelTest : RobolectricTest() {
+      ActionConfig(
+        trigger = ActionTrigger.ON_CLICK,
+        workflow = ApplicationWorkflow.LAUNCH_QUESTIONNAIRE,
+        params =
+          listOf(
+            ActionParameter(
+              paramType = ActionParameterType.PREPOPULATE,
+              linkId = "25cc8d26-ac42-475f-be79-6f1d62a44881",
+              dataType = DataType.INTEGER,
+              key = "maleCondomPreviousBalance",
+              value = "100"
+            )
+          )
+      )
+    val overflowMenuItemConfig =
+      OverflowMenuItemConfig(visible = "", actions = listOf(actionConfig))
 
-    val slot = slot<Bundle>()
-    verify {
-      (context as QuestionnaireHandler).launchQuestionnaire<Any>(
-        context = context,
-        intentBundle = capture(slot),
-        questionnaireConfig = actionConfig.questionnaire,
-        actionParams = actionConfig.params
+    val event =
+      ProfileEvent.OverflowMenuClick(
+        navController = navController,
+        resourceData = null,
+        overflowMenuItemConfig = overflowMenuItemConfig,
       )
+
+    runBlocking {
+      profileViewModel.onEvent(event)
+      delay(100)
     }
 
-    val capturedIntentBundle = slot.captured
-    assertNotNull(capturedIntentBundle.getString(QuestionnaireActivity.QUESTIONNAIRE_RESPONSE))
+    Assert.assertNotNull(profileViewModel.snackBarStateFlow)
+    coVerify(inverse = true) { registerRepository.fhirEngine }
   }
 
   @Test
-  fun testOnEventLaunchQuestionnaireWhenQuestionnaireResponseFromDbIsNotFoundShouldReturnNullQuestionnaireResponse() {
+  @kotlinx.coroutines.ExperimentalCoroutinesApi
+  fun testShouldHaveErrorSnackBarMessageWhenQuestionnaireIsNull() {
+    val context = mockk<Context>(moreInterfaces = arrayOf(QuestionnaireHandler::class))
+    val navController = NavController(context)
+    val actionConfig =
+      ActionConfig(
+        trigger = ActionTrigger.ON_CLICK,
+        workflow = ApplicationWorkflow.LAUNCH_QUESTIONNAIRE,
+        questionnaire = QuestionnaireConfig(id = "444", type = QuestionnaireType.EDIT),
+        params =
+          listOf(
+            ActionParameter(
+              paramType = ActionParameterType.PREPOPULATE,
+              linkId = "25cc8d26-ac42-475f-be79-6f1d62a44881",
+              dataType = DataType.INTEGER,
+              key = "maleCondomPreviousBalance",
+              value = "100"
+            )
+          )
+      )
+    val overflowMenuItemConfig =
+      OverflowMenuItemConfig(visible = "", actions = listOf(actionConfig))
+
+    coEvery { registerRepository.fhirEngine.get(ResourceType.Questionnaire, "444") } throws
+      ResourceNotFoundException("type", "id")
+
+    val event =
+      ProfileEvent.OverflowMenuClick(
+        navController = navController,
+        resourceData = null,
+        overflowMenuItemConfig = overflowMenuItemConfig,
+      )
+
+    runBlocking {
+      profileViewModel.onEvent(event)
+      delay(100)
+    }
+
+    Assert.assertNotNull(profileViewModel.snackBarStateFlow)
```

Comment:
I added this, but it would be better to test the actual error in the snackBarState, work for CHT

Reply:
yep, noted